File: backend/app/models/comments.py
"""Pydantic models for comment endpoints."""

from typing import Optional
from pydantic import BaseModel


class CreateCommentBody(BaseModel):
    entity_type: str
    entity_id: str
    parent_id: Optional[str] = None
    author: str = "user"
    body: str
    mentions: list[str] = []


class PatchCommentBody(BaseModel):
    body: Optional[str] = None
    mentions: Optional[list[str]] = None

File: frontend/src/types/comments.ts
export interface Comment {
  id: string;
  entity_type: string;
  entity_id: string;
  parent_id: string | null;
  author: string;
  body: string;
  mentions: string; // JSON-encoded string array
  created_at: string;
  updated_at: string;
}

export interface CreateCommentPayload {
  entity_type: string;
  entity_id: string;
  parent_id?: string | null;
  author?: string;
  body: string;
  mentions?: string[];
}

export interface UpdateCommentPayload {
  body?: string;
  mentions?: string[];
}

export interface MentionOption {
  id: string;
  label: string;
  type: 'person' | 'agent';
}
